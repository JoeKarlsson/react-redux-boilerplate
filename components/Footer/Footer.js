import React from 'react';
import Link from '../shared/Link';

const Footer = () => (
  <div>
    <h2>
      Footer
    </h2>
  </div>
);

export default Footer;
