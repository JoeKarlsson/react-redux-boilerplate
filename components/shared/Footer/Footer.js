import React from 'react';
import styles from './Footer.scss';

const Footer = () => (
  <div className={styles.footer}>
    <h2>
      Footer
    </h2>
  </div>
);

export default Footer;
