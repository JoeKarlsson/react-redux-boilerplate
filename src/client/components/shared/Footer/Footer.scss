.footer {
	width: 100%;
	padding-top: 1%;
	padding-bottom: 1%;

	.footer_content {
		text-align: center;
	}
}
