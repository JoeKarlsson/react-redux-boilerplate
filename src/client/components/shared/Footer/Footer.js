import React from 'react';
import './Footer.scss';

const Footer = () => (
	<div className="footer">
		<h2>
			<div className="footer_content" >
				<p>
					site by <a target="_blank" rel="noopener noreferrer" href="http://www.callmejoe.net/">joe</a>
				</p>
			</div>
		</h2>
	</div>
);

export default Footer;
