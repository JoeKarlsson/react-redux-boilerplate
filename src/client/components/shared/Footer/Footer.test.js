import React from 'react';
import {
	shallow,
	configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Footer from './Footer';

configure({ adapter: new Adapter() });

describe('Footer', () => {
	let wrapper;

	global.requestAnimationFrame = (callback) => {
		setTimeout(callback, 0);
	};

	beforeEach(() => {
		wrapper = shallow(<Footer />);
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('is rendered', () => {
				const component = renderer.create(<footer />);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});

			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});

			it('should render without throwing an error', () => {
				expect(wrapper.contains(
					<div className="footer">
						<h2>Footer</h2>
					</div>,
				)).toEqual(true);
			});

			it('should be selectable by the class `footer`', () => {
				expect(wrapper.is('.footer')).toBe(true);
			});

			it('should mount in the full DOM', () => {
				expect(wrapper.find('.footer').length).toBe(1);
			});

			it('should render to static HTML', () => {
				expect(wrapper.text()).toEqual('Footer');
			});

			it('should have correct inital instance', () => {
				const initialInstance = wrapper.instance();
				const expectedInstance = null;
				expect(initialInstance).toBe(expectedInstance);
			});
		});
	});
});
