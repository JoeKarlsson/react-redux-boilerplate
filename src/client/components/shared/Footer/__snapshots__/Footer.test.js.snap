// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer rendering initial state is rendered 1`] = `
<div
  className="footer"
>
  <h2>
    <div
      className="footer_content"
    >
      <p>
        site by 
        <a
          href="http://www.callmejoe.net/"
          rel="noopener noreferrer"
          target="_blank"
        >
          joe
        </a>
      </p>
    </div>
  </h2>
</div>
`;
